assests/JavaScript/logic.js

```diff
@@ -0,0 +1,225 @@
+let correct = 0
+let incorrect = 0
+let unanswered = 0
+let index = 0
+let randomMovieArray = []
+let moviePool = movieDataBase.length
+let status = true
+let m = -1
+let timer = setInterval
+
+//which game is selected
+let newGame = decodeURIComponent(window.location.search);
+newGame = newGame.substring(1);
+let finalGame = newGame.replace('para1=','');
+if (finalGame === "what") {
+    $('#curtain1').animate({width: '260px'}, 2000)
+    $('#curtain2').animate({width: '260px'}, 2000)
+    gameSetUp()
+    setTimeout (playWhatMovie, 2500)
+} else {
+    $('#curtain1').animate({width: '260px'}, 2000)
+    $('#curtain2').animate({width: '260px'}, 2000)
+    gameSetUp()
+    setTimeout (playWhoSaid, 2500)
+}
+
+function gameSetUp () {
+    //randomly pick 10 movies for quiz
+    for (i = 0; i < 10; i++) {
+        let movieNumber = Math.floor(Math.random()*(moviePool))
+        if (randomMovieArray.indexOf(movieNumber) === -1) {
+            randomMovieArray.push(movieNumber)
+        } else {
+            i--
+        }
+    }
+    //get info from 10 random movies
+    for (i = 0; i < randomMovieArray.length; i++) {
+        let index = randomMovieArray[i]
+    }
+}
+
+//plays what movie
+function playWhatMovie () { 
+    if (index < randomMovieArray.length) {
+        m = randomMovieArray[index]
+        $('.movie-answer').html(`
+            <div class = "question">
+            <h2 id = "in-which-movie">In which movie is it said:</h2>
+            <h2 id ="question-movie-line">"${movieDataBase[m].line}"</h2>
+            <div class="choice-container"></div>  
+            </div>
+            `)
+        whatMovieOptions()
+        setTimer (playWhatMovie) 
+    } else {
+        endGame()
+    }
+}
+
+//plays who said
+function playWhoSaid (){
+    if (index < randomMovieArray.length) {
+        m = randomMovieArray[index]
+        $('.movie-answer').html(`
+            <div class = "question">
+            <h2 id = "in-which-movie">Who said:</h2>
+            <h2 id ="question-movie-line">"${movieDataBase[m].line}"</h2>
+            <div class="choice-container"></div> 
+            </div>
+            `)
+        whoSaidOptions()
+        setTimer (playWhoSaid) 
+    } else {
+        endGame()
+    }
+}
+
+//sets timer
+function setTimer (game) {
+    let t = 10
+    timer = setInterval (function () {
+        $("#timer").html(t + " seconds")
+        t--
+        if (t < 0) {
+            unanswered++
+            clearInterval(timer)
+            showAnswer('', game)
+        }
+    }, 1000)
+}
+
+//create random choice options/makes sure no duplicates even though speaker in data base more than once
+function whatMovieOptions () {
+    let choiceOptions = []
+    let optionsString = []
+    choiceOptions.push(m)
+    optionsString.push(movieDataBase[m].title)
+    for (i=0; i < 3; i++) {
```

Comment:
Because you did not use `let i...` or `var i...`, the variable `i` is now in the global scope and I can access it via the Google Chrome DevTools. Easy to miss the mistake, but make sure you initialize and declare your variables properly.